quarkchain/tools/check_health_bootstrap.py

```diff
@@ -0,0 +1,135 @@
+
+will update BOOT_NODES fields in P2P section and check the working status of boostrap nodes.
+"""
+import argparse
+import asyncio
+import logging
+import time
+from datetime import datetime
+import jsonrpcclient
+import psutil
+import numpy
+from decimal import Decimal
+import smtplib
+from quarkchain.cluster.cluster_config import ClusterConfig
+from quarkchain.cluster.master import MasterServer
+from quarkchain.cluster.cluster import Cluster
+import jsonrpcclient
+import logging
+import time
+from datetime import datetime
+import smtplib
+import os
+
+TIMEOUT = 10
+PRIVATE_ENDPOINT = "http://{}:38491".format("localhost")
+PRIVATE_CLIENT = jsonrpcclient.HTTPClient(PRIVATE_ENDPOINT)
+
+# Please export your email information first.
+# export EMAIL_FROM_ADDRESS=""
+# export PASSWORD=""
+# export EMAIL_TO_ADDRESS=""
+
+
+def now():
+    return datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+
+class Health_Check_Cluster(Cluster):
+    async def run(self):
+        await self.run_master()
+        # p2p discovery mode will disable slaves
+        # if not self.config.P2P.DISCOVERY_ONLY:
+        #    await self.run_slaves()
+
+        await asyncio.sleep(20)
+        checkRoutingTable(10)
+        await self.shutdown()
+
+
+def checkRoutingTable(timeout=TIMEOUT):
+    result = PRIVATE_CLIENT.send(
+        jsonrpcclient.Request("getKadRoutingTable"), timeout=timeout
+    )
+    if len(result) == 0:
+        print("Bootstrap node can not provide the routing table for a while!")
+        subject = "Boostrap Node Alert!"
+        msg = "Bootstrap node can not provide the routing table for a while!" + now()
+        send_email(subject, msg)
+    print(len(result))
+
+
+def send_email(subject, msg):
+    try:
+        server = smtplib.SMTP("smtp.gmail.com:587")
+        server.ehlo()
+        server.starttls()
+        if "EMAIL_FROM_ADDRESS" in os.environ:
+            EMAIL_FROM_ADDRESS = os.environ.get("EMAIL_FROM_ADDRESS")
+        if "PASSWORD" in os.environ:
+            PASSWORD = os.environ.get("PASSWORD")
+        if "EMAIL_TO_ADDRESS" in os.environ:
+            EMAIL_TO_ADDRESS = os.environ.get("EMAIL_TO_ADDRESS")
+        server.login(EMAIL_FROM_ADDRESS, PASSWORD)
+        message = "Subject: {}\n\n{}".format(subject, msg)
+        server.sendmail(EMAIL_FROM_ADDRESS, EMAIL_TO_ADDRESS, message)
+        server.quit()
+        print("Success: Email sent!")
+    except:
+        print("Email failed to send.")
+
+
+def main():
+    # os.chdir(os.path.dirname(os.path.abspath(__file__)))
+    os.chdir(os.path.dirname("../../quarkchain/cluster/"))
+    parser = argparse.ArgumentParser()
+    ClusterConfig.attach_arguments(parser)
+    args = parser.parse_args()
+
+    config = ClusterConfig.create_from_args(args)
+    print("Cluster config file: {}".format(config.json_filepath))
+    print(config.to_json())
+
+    cluster = Health_Check_Cluster(config)
+    bootnodes = []
+    # GCP Bootstrap
+    bootnodes.append(
```

Comment:
how about just read through `config.P2P.BOOTNODES`?

Reply:
Since config_p2p.py is used to reset the bootnode's address, the config of P2P bootnodes are not fixed based on this tool.

Reply:
then I would suggest using another file as config for health check, which should be a copy of the original config file